packages/react/src/keyframes/__tests__/index.test.tsx

```diff
@@ -2,20 +2,28 @@ import { keyframes, styled } from '@compiled/react';
 import { render } from '@testing-library/react';
 import * as React from 'react';
 
+import { resetStyleBuckets } from '../../runtime/sheet';
+import { resetCache } from '../../runtime/style-cache';
 import defaultFadeOut, { namedFadeOut, fadeOut as shadowedFadeOut } from '../__fixtures__';
 
 const getOpacity = (str: string | number) => str;
 
 const getKeyframe = (name: string) => {
   const styles = Array.from(
-    document.body.querySelectorAll('style'),
+    document.head.querySelectorAll('style'),
     (style) => style.innerHTML
   ).join('\n');
 
   return styles.substring(styles.indexOf(`@keyframes ${name}`));
 };
 
 describe('keyframes', () => {
+  afterEach(() => {
+    resetStyleBuckets();
```

Comment:
Can't these helpers be removed now that the `browser` build is mostly reverted?

Reply:
No, the helpers are necessary because these tests now believe they're running in the browser. Previously, all tests ran in SSR mode unless the `isNodeEnvrionment` helper was explicitly mocked to `false`  (because `process` exists even in JSDOM environments - just doesn't try to hide it in any way).

I can switch this test to run in SSR mode again by changing the Jest environment for it, however it seems like these tests are meant to test browser behaviour since there's no mention of SSR in here.

Reply:
Okay, let me have a look to see if there's a better way of tackling this

Reply:
Using environment variables seems better here, since it can be stripped out in the builds...

In the `ts-transform-define`, add:

```
"replace": {
  "isNodeEnvironment()": "process.env.IS_NODE_EXPRESSION",
  "process.env.NODE_ENV === 'test'": "false"
}
```

Then make these changes:

```
const Cache: any = isNodeEnvironment() ? createContext<Record<string, true> | null>(null) : {};

export const useCache: UseCacheHook = () => {
  if (process.env.NODE_ENV === 'test' && process.env.CACHE === 'false') {
    return {};
  }

  // ...
};
```

```
const styleBucketsInHead: Partial<Record<Bucket, HTMLStyleElement>> = {};

function lazyAddStyleBucketToHead(bucketName: Bucket, opts: StyleSheetOpts): HTMLStyleElement {
  if (!styleBucketsInHead[bucketName] || (process.env.NODE_ENV === 'test' && process.env.CACHE === 'false')) {
    // ...
 }

 return styleBucketsInHead[bucketName]!;
};
```

In the test:

```
beforeAll(() => {
  process.env.CACHE = 'false';
});

afterAll(() => {
  delete process.env.CACHE;
});
```
I tried a bunch of other ways with jest mocks, isolated modules, separate test utils entrypoint, but to no avail.

Reply:
> It might be better putting the check in its own util, as that could be stripped out more easily. Not sure if it impacts the bundlesize much though.

I can't actually get `ts-transform-define` to work with just the check just written out inline, so looks like I have to go with the function anyway. Looks like in the end, it only adds 1B to the bundle size that we check for.

Reply:
Awesome, 1B is fine! If we move to a proper bundler later on, we can tree-shake it properly.